frontend/webapp/hooks/instrumentation-rules/useInstrumentationRuleCRUD.ts

```diff
@@ -48,33 +39,30 @@ export const useInstrumentationRuleCRUD = (): UseInstrumentationRuleCrud => {
     } else if (data?.computePlatform?.instrumentationRules) {
       const { instrumentationRules: items } = data.computePlatform;
 
-      addEntities(ENTITY_TYPES.INSTRUMENTATION_RULE, mapFetched(items));
+      addEntities(ENTITY_TYPES.INSTRUMENTATION_RULE, items);
       setEntitiesLoading(ENTITY_TYPES.INSTRUMENTATION_RULE, false);
     }
   };
 
-  const [mutateCreate, cState] = useMutation<{ createInstrumentationRule: FetchedInstrumentationRule }, { instrumentationRule: InstrumentationRuleFormData }>(CREATE_INSTRUMENTATION_RULE, {
+  const [mutateCreate, cState] = useMutation<{ createInstrumentationRule: InstrumentationRule }, { instrumentationRule: InstrumentationRuleFormData }>(CREATE_INSTRUMENTATION_RULE, {
     onError: (error) => notifyUser(STATUS_TYPE.ERROR, error.name || CRUD.CREATE, error.cause?.message || error.message),
     onCompleted: (res) => {
       const rule = res.createInstrumentationRule;
       const type = deriveTypeFromRule(rule);
-      addEntities(ENTITY_TYPES.INSTRUMENTATION_RULE, mapFetched([rule]));
+      addEntities(ENTITY_TYPES.INSTRUMENTATION_RULE, [rule]);
       notifyUser(STATUS_TYPE.SUCCESS, CRUD.CREATE, `Successfully created "${type}" rule`, rule.ruleId);
     },
   });
 
-  const [mutateUpdate, uState] = useMutation<{ updateInstrumentationRule: FetchedInstrumentationRule }, { ruleId: string; instrumentationRule: InstrumentationRuleFormData }>(
-    UPDATE_INSTRUMENTATION_RULE,
-    {
-      onError: (error) => notifyUser(STATUS_TYPE.ERROR, error.name || CRUD.UPDATE, error.cause?.message || error.message),
-      onCompleted: (res) => {
-        const rule = res.updateInstrumentationRule;
-        const type = deriveTypeFromRule(rule);
-        addEntities(ENTITY_TYPES.INSTRUMENTATION_RULE, mapFetched([rule]));
-        notifyUser(STATUS_TYPE.SUCCESS, CRUD.UPDATE, `Successfully updated "${type}" rule`, rule.ruleId);
-      },
+  const [mutateUpdate, uState] = useMutation<{ updateInstrumentationRule: InstrumentationRule }, { ruleId: string; instrumentationRule: InstrumentationRuleFormData }>(UPDATE_INSTRUMENTATION_RULE, {
+    onError: (error) => notifyUser(STATUS_TYPE.ERROR, error.name || CRUD.UPDATE, error.cause?.message || error.message),
+    onCompleted: (res) => {
+      const rule = res.updateInstrumentationRule;
+      const type = deriveTypeFromRule(rule);
```

Comment:
we can use rule.type now right?
```suggestion
      const type = rule.type
```